autogpt_platform/backend/backend/data/expert_spend.py

```
@@ -0,0 +1,76 @@
+            f"{type(e).__name__}: {e}"
+        )
+
+
+async def get_weekly_spend(expert_id: str) -> int:
+    """Current-week spend in credits; 0 on any read failure (errs open —
+    the budget gate must not block runs because Redis hiccuped)."""
+    try:
+        redis = await get_redis_async()
+        value = await redis.get(weekly_spend_key(expert_id))
+        return int(value) if value is not None else 0
+    except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)